ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/BlockValueCalculator.java

```diff
@@ -27,7 +27,13 @@ public Wei calculateBlockValue(final BlockWithReceipts blockWithReceipts) {
     Wei totalFee = Wei.ZERO;
     for (int i = 0; i < txs.size(); i++) {
       final Wei minerFee = txs.get(i).getEffectivePriorityFeePerGas(block.getHeader().getBaseFee());
-      totalFee = totalFee.add(minerFee.multiply(receipts.get(i).getCumulativeGasUsed()));
+      // we don't store gasUsed and need to calculate that on the fly
```

Comment:
Can the TransactionWithMetadata type be used here? If we can use that then there is getGasUsed method that can be used instead to avoid duplicating this logic. Should be able to use the BlockchainQueries.blockByHash from the EngineGetPayloadV2 method I think.

Reply:
After some discussion, the BlockchainQueries can't be used as these blocks aren't appended yet. So this approach isn't possible